src/main/kotlin/com/amazon/ionelement/impl/IonElementLoaderImpl.kt

```diff
@@ -89,18 +89,13 @@ internal class IonElementLoaderImpl(private val options: IonElementLoaderOptions
         return handleReaderException(ionReader) {
             val valueType = requireNotNull(ionReader.type) { "The IonReader was not positioned at an element." }
 
-            val annotations = ionReader.typeAnnotations!!.asList().toEmptyOrPersistentList()
+            val annotations = ionReader.typeAnnotations!!.toImmutableListUnsafe()
 
-            val metas = when {
-                options.includeLocationMeta -> {
-                    val location = ionReader.currentLocation()
-                    when {
-                        location != null -> metaContainerOf(ION_LOCATION_META_TAG to location)
-                        else -> emptyMetaContainer()
-                    }
-                }
-                else -> emptyMetaContainer()
-            }.toPersistentMap()
+            var metas = EMPTY_METAS
+            if (options.includeLocationMeta) {
+                val location = ionReader.currentLocation()
+                if (location != null) metas = location.toMetaContainer()
+            }
```

Comment:
🗺️ This is partly a stylistic change. The `toMetaContainer()` wraps the `IonLocation` in an `ImmutableMap` implementation, which is the main performance improvement for the location metadata. The rest of this change is stylistic to try to make it more readable and concise.